offer-manager-domain/src/main/groovy/life/qbic/business/offers/Offer.groovy

```diff
@@ -504,7 +474,9 @@ class Offer {
     }
 
     private BigDecimal calculateTotalDiscountAmount() {
-        return items.sum{ BigDecimal.valueOf(it.quantityDiscount) } as BigDecimal
+        return items.stream()
+                .map({ BigDecimal.valueOf(it.quantityDiscount) })
+                .reduce(BigDecimal.ZERO, BigDecimal::add)
```

Comment:
elegant functional programming, though not as readable for everyone ;) I like 👍🏽